src/safir/testing/kubernetes.py

```diff
@@ -44,6 +52,24 @@
 ]
 
 
+def _stringify_selector_dict(selector: dict[str, str]) -> str:
+    """This and the following function transmute selector labels between
+    dicts and strings.  We don't even try to do safe quoting."""
+    sstrings: list[str] = []
+    for key in selector:
+        sstrings.append(f"{key}=={selector[key]}")
+    return ",".join(sstrings)
+
+
+def _dictify_selector_str(selector_str: str) -> dict[str, str]:
+    sstrings = selector_str.split(",")
+    retval: dict[str, str] = {}
+    for sel in sstrings:
+        (key, val) = sel.split("==", 1)
+        retval[key] = val
+    return retval
```

Comment:
While dictify is more specific, it's an awkward term. I think this is just parsing, so I'd call this `_parse_label_selector`. Would be good to add a proper docstring and comment that it only supports equality (and only `==`, not `=`, although ideally we would support both since Kubernetes does). Kubernetes calls each stanza a "requirement."
```suggestion
def _parse_label_selector(label_selector: str) -> dict[str, str]:
    """Parse a label selector.
    
    Only equality is supported (with ``=`` or ``==``).
    
    Parameters
    ----------
    label_selector
        Label selector string to parse.
        
    Returns
    ------
    dict of str to str
        Dictionary of required labels to their required values.
        
    Raises
    ------
    AssertionError
        Raised if the label selector string is not a supported syntax.
    """
    result = {}
    for requirement in label_selector.split(","):
        match = re.match(r"([^!=]+)==?(.*)", requirement)
        assert match and match.group(1) and match.group(2)
        result[match.group(1)] = match.group(2)
    return result
```